pkg/daemon/file_writers.go

```diff
@@ -60,6 +61,12 @@ func createOrigFile(fromPath, fpath string) error {
 		return writeFileAtomicallyWithDefaults(noOrigFileStampName(fpath), nil)
 	}
 
+	// https://issues.redhat.com/browse/OCPBUGS-11437
+	// MCO keeps the pull secret to .orig file once it replaced
+	if strings.HasSuffix(fpath, "config.json") {
```

Comment:
I think like Sergio mentions, this probably should check specifically for the pull secret path and not just any config.json

Reply:
Thanks, @sergiordlr and @yuqi-zhang for pointing this out. I have made the changes to more carefully point at the desired path, and yet to be tested.